usb/imx6ull-usbc/imx6ull-usbclient.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Phoenix-RTOS
+ *
+ * usbclient - usb device controller driver
+ *
+ * Copyright 2019 Phoenix Systems
+ * Author: Kamil Amanowicz, Bartosz Ciesla, Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "../imx-usbc/imxdevice.h"
+
+#include <sys/interrupt.h>
+#include <sys/threads.h>
+#include <sys/debug.h>
+#include <sys/mman.h>
+#include <sys/msg.h>
+#include <string.h>
+#include <errno.h>
+
+#define USB_ADDR 0x02184000
+
+
+static usb_common_data_t usb_data;
+static usb_dc_t dc = { 0 };
+
+
+static int ctrlqh_init(void)
+{
+	uint32_t qh_addr;
+
+	/* map queue head list */
+	dc.endptqh = mmap(NULL, USB_BUFFER_SIZE, PROT_WRITE | PROT_READ, MAP_UNCACHED, OID_NULL, 0);
```

Comment:
mmap works only with multiples of _PAGE_SIZE. USB_BUFFER_SIZE is equal to _PAGE_SIZE on imx6ull by *accident*,  and is multiple of _PAGE_SIZE (which is 0x200) and imxrt by *accident*. It would be better if it wasn't coincidental.